framework/include/base/Moose.h

```diff
@@ -38,12 +38,18 @@ void MooseMatView(SparseMatrix<Number> & mat);
 void MooseMatView(const SparseMatrix<Number> & mat);
 
 /**
- * MOOSE now contains C++11 code, so give a reasonable error message
- * stating the minimum required compiler versions.
+ * MOOSE now contains C++17 code, so give a reasonable error message
+ * stating what the user can do to address this in their environment if C++17
+ * compatibility isn't found.
  */
-#ifndef LIBMESH_HAVE_CXX11
-#error MOOSE requires a C++11 compatible compiler (GCC >= 4.8.4, Clang >= 3.5.1). Please update your compiler and try again.
-#endif
+namespace Moose
+{
+static_assert(__cplusplus >= 201703L,
+              "MOOSE requires a C++17 compatible compiler (GCC >= 7.5.0, Clang >= 5.0.2). Please "
+              "update your compiler or, if compatible, add '-std=c++17' to your compiler flags "
+              "and try again. If using the MOOSE conda package, please create a new MOOSE "
+              "environment to ensure an update to the newest available package.");
```

Comment:
I dont think conda users need a new environment. They just need to update, and if unsuccessful, create a new environment. Yaqi, myself and at least one other only had to update

Reply:
To many users have needed to wipe their env

Reply:
How about providing a script to wipe and rebuild the env?

Reply:
@GiudGiud I updated the message to include both options, with update being priority. 

@dschwen Not a bad idea - though we cannot know beforehand what name they chose for their moose environment, so I would prefer pointing them to some instructions they can tailor if needed. We have website instructions that outline how to wipe and get started again, so I added that URL to the message to make it easier.